src/Compilers/Core/Portable/FileSystem/IFileSystem.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+
+namespace Roslyn.Utilities
+{
+    /// <summary>
+    /// Abstraction over the file system that is useful for test hooks
+    /// </summary>
+    internal interface IFileSystem
+    {
+        bool FileExists(string filePath);
+
+        Stream OpenFile(string filePath, FileMode mode, FileAccess access, FileShare share);
+
+        Stream OpenFileEx(string filePath, FileMode mode, FileAccess access, FileShare share, int bufferSize, FileOptions options, out string normalizedFilePath);
```

Comment:
>OpenFileEx [](start = 15, length = 10)

Is "Ex" necessary? Why not just an overload?

Reply:
I was attempting to have the name of the functions match up with the delegate names and hook names test layer. If you diff by overload then the functions you add in the test layer don't line up anymore. I don't feel strongly about this but I do find it visually appealing and simple in the consumption side.